transcom/mymove
# MB-13922: show payment requests docs

## [Jira ticket](https://dp3.atlassian.net/browse/MB-13922)

## Summary

Prior to these changes we were showing the orders docs when reviewing weights. Instead of showing that we wanted to show all of the documents associated with a payment request.

Note: Since a single move can have multiple payment requests we need to show all of the docs across all of the payment requests. (At least that is my understanding.)

Please pay extra attention to the following:
- Are there edge cases where you can review the weights before there is a payment request? 
- Are there some edge cases that I missed where the array functions here would fail?
- Is there a way to simplify this code?
- Are there any react query standards or conventions that I am breaking? I'm not very familiar with this library so please call out any oddities or poor usages.

You may still occasionally see a console warning about a React state update on an unmounted component. This is an existing issue and I'll be creating a separate [bug](https://dp3.atlassian.net/browse/MB-16611) for this.
Eg. 
<img width="895" alt="image" src="https://github.com/transcom/mymove/assets/84742904/30cc8e4d-095f-41cb-9363-5b6b3335b69e">


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

Create a move and progress it to the prime and create a payment request and then check docs as a TIO. You can at each step also try to visit the payment requests page and try and interact with the review weight button to be thorough. 

Alternatively use testharness.

1. visit the `/testharness/list` page. (eg. http://officelocal:3000/testharness/list)
2. Click on the `HHGMoveNeedsSC` button to create a move and copy its moveCode.
3. Login as the SC and find the move
4. Approve the move as the SC, this will send the move to the TOO
5. Login as the TOO and add the shipments and the move management. This will make the move available to the prime
6. Login as the prime and add a payment request and upload docs to the payment request.
7. Login as the TIO and try and review the weights. You should see all of the uploaded docs.
